### ISSUE TITLE
Invalid Head/Pitch Rotations (< -90 & > 90)

### ISSUE THREAD

**Issue by ghost** [NoCheatPlus/Issues#473]
#### Short description of the issue.
No visible checks for capping/clamping an invalid/incapable rotational pitch value (< -90 and > 90) to -90 and 90 besides flagging FIGHT/BLOCK_DIRECTION

#### Complete output of the _ncp version_ command (please prefer text here).
---- Version information ----
#### Server ####
git-Spigot-21fe707-e1ebe52 ~MC: 1.8.8~
detected: 1.8.8
#### NoCheatPlus ####
Plugin: 3.15.1-RC-sMD5NET-b1084
MCAccess: 1.8.4-1.8.8 / Spigot-CB-1.8_R3
Features:
 blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8
checks: FastConsume | Gutenberg | AttackFrequency | FlyingFrequency | KeepAliveFrequency
defaults: pvpKnockBackVelocity
packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance | CatchAllAdapter
Hooks: AllViolations~NCP~ 1.0
#### Related Plugins ####
ProtocolLib v4.3.0

#### How to reproduce the issue (is it random / always happens / side conditions)?
Grab a hacked client that has Headless or Derp with Headless option

#### Have you made any (noteworthy) changes to the default configuration of NCP?
No

#### Do you run any other special plugins that affect game mechanics next to NC+ (adding skills, machines, adding/changing blocks, other anti cheating)?
No, but I did make a separate plugin to patch the invalid pitch myself

**Comment by Janmm14** [comment]
You can give enemies reversed velocity through this.

**Comment by ghost** [comment]
@Janmm14 yes I forgot to mention that glitch (visible I believe in the Direkt hacked client) where it will quickly turn your head before (or after cant remember) you attack to the reverse angle and throw the entity possibly towards you or in whatever direction you please (sort of like as if your player had a gravitation pull)

**Comment by RoboMWM** [comment]
Duplicate of https://github.com/NoCheatPlus/Issues/issues/77 though I have been having second thoughts about this, since I initially thought it's just one of many ways a player clearly exposes that they're using a cheat mod (and thus probably better suited for something exclusively looking for that, since there's no reason to prevent)... However, I guess in the case of it being used in combat might be something worth checking and canceling for.

**Comment by asofold** [comment]
Added a Bukkit based check. I assume it can be checked on Bukkit side - if not, please note.
